test/framework/src/main/java/org/elasticsearch/index/mapper/TestDocumentParserContext.java

```diff
@@ -61,7 +61,9 @@ private TestDocumentParserContext(MappingLookup mappingLookup, SourceToParse sou
                 MapperTestCase.createIndexSettings(Version.CURRENT, Settings.EMPTY),
                 null
             ),
-            source
+            source,
+            mappingLookup.getMapping().getRoot(),
+            ObjectMapper.Dynamic.getRootDynamic(mappingLookup)
```

Comment:
this made me wonder if this context should extend from the root parser context, but that may be such a trivial change...

Reply:
I was wondering if we could nuke it altogether, but we can look into that later I think.